### PR TITLE
Feature - add User Secrets secret provider to secret store

### LINKED ISSUE #152: Secret Provider for ASP.NET SecretManager
**What secret store are you interested in?**
A Secret Provider for the ASP.NET Secret Manager.

**What scenarios would it enable you?**

In ASP.NET, there exists a secret manager which can be used to store secrets while developing.  
Normally, this secret manager is added as a configuration provider, but if you want to separate secrets from config, I think Arcus should provide  Secret Provider for the asp.net secret manager.

**More information**
More information regarding the secret provider and how it works can be found [here](https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-3.1&tabs=windows#secret-manager)

### PR DESCRIPTION
**Pull request by stijnmoreels**
Closes #152

### CODE REVIEW

File: src/Arcus.Security.Providers.UserSecrets/Extensions/SecretStoreBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            var source = new JsonConfigurationSource
+            {
+                FileProvider = fileProvider,
+                Path = SecretsFileName,
+                ReloadOnChange = reloadOnChange,
+                Optional = false
+            };
+
+            source.ResolveFileProvider();
+            if (source.FileProvider == null)
```
Comment (stijnmoreels) [inline src/Arcus.Security.Providers.UserSecrets/Extensions/SecretStoreBuilderExtensions.cs:114]: This extra check and default value is something I saw in the original source, so I added it here too.

**Comment by stijnmoreels** [comment]
Will add some more tests.

File: docs/preview/features/secret-store/provider/user-secrets.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        CreateHostBuilder(args).Build().Run();
+    }
+
+    public static IHostBuilder CreateHostBuilder(string[] args)
+    {    
+        return Host.CreateDefaultBuilder(args)
+                   .ConfigureSecretStore((context, config, builder) =>
+                   {
+                         // Adds the user secrets configuration source with specified user secrets ID.
```
Comment (tomkerkhove) [inline docs/preview/features/secret-store/provider/user-secrets.md:34]: This is more accurate maybe?
```suggestion
                         // Adds the user secrets secret source with specified user secrets ID.
```

File: docs/preview/features/secret-store/provider/user-secrets.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                         // A user secrets ID is unique value used to store and identify a collection of secrets.
```
Comment (tomkerkhove) [inline docs/preview/features/secret-store/provider/user-secrets.md:35]:
```suggestion
                         // A user secrets ID is a unique value used to store and identify a collection of secrets.
```

File: src/Arcus.Security.Providers.UserSecrets/Arcus.Security.Providers.UserSecrets.csproj [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;netcoreapp3.1</TargetFrameworks>
+    <Authors>Arcus</Authors>
+    <Description>Provides support for User Secrets</Description>
```
Comment (tomkerkhove) [inline src/Arcus.Security.Providers.UserSecrets/Arcus.Security.Providers.UserSecrets.csproj:6]: Just a suggestion
```suggestion
    <Description>Provides support for User Secrets with Arcus Secret Store</Description>
```

**Pull request merged**